.github/workflows/test.yml

```diff
@@ -38,3 +38,26 @@ jobs:
         shell: bash -l {0}
         run: |
           coverage report
+
+  mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
+          auto-activate-base: false
+
+      - name: Install dependencies
+        shell: bash -l {0}
+        run: |
+          conda install python=3.11 mypy pyyaml graphviz
+          pip install types-PyYAML
+
+      - name: Run mypy
+        shell: bash -l {0}
+        run: |
+          mypy .
```

Comment:
Also, Instead of mypy specific workflow, maybe we could simply have a `pre-commit` workflow which would check the entirety of pre-commit.

Reply:
Done. I've also enabled strict mode in mypy